lib/awshark/cloud_formation/template.rb

```diff
@@ -28,16 +30,20 @@ def body
 
       # @returns [Hash]
       def context
-        @context ||= begin
-                       context = load_file(context_path) || {}
-                       context = context[stage] if context.key?(stage)
-
-                       {
-                         context: RecursiveOpenStruct.new(context),
-                         aws_account_id: Awshark.config.aws_account_id,
-                         stage: stage
-                       }
-                     end
+        @context ||=
+          begin
+            context = load_file(context_path) || {}
+            context = context[stage] if context.key?(stage)
+
+            {
+              context: RecursiveOpenStruct.new(context),
+              aws_account_id: Awshark.config.aws_account_id,
+              stage: stage,
+              ssm: Proc.new do |key|
+                ssm_client.get_parameter(name: key, with_decryption: true)&.parameter&.value
+              end
+            }
```

Comment:
Wouldn't this also work and might look simpler?

```
                       {
                         context: RecursiveOpenStruct.new(context),
                         aws_account_id: Awshark.config.aws_account_id,
                         stage: stage,
                         ssm: ssm
                       }
                     end                       
   ...

    def ssm
       @ssm ||= Aws::SSM::Client.new
       Proc.new do |key|
         @ssm.get_parameter(name: key, with_decryption: true)&.parameter&.value
       end
    end
```

Reply:
really, more clear code.